xbmc/weather/WeatherManager.cpp

```diff
@@ -95,21 +96,31 @@ std::string CWeatherManager::GetLocation(int iLocation) const
 
 void CWeatherManager::Reset()
 {
+  std::unique_lock lock(m_critSection);
   m_info = {};
 }
 
 bool CWeatherManager::IsFetched()
 {
   // call GetInfo() to make sure that we actually start up
   GetInfo(0);
+
+  std::unique_lock lock(m_critSection);
   return !m_info.lastUpdateTime.empty();
 }
 
 const ForecastDay& CWeatherManager::GetForecast(int day) const
 {
+  std::unique_lock lock(m_critSection);
```

Comment:
I don't think this lock servers a purpose here. Returning the reference is independent of anyone reading or writing `m_info.forecast`. Possible races will happen when the returned reference is used, but this can't be resolved with only internal locks.

One option to resolve this is by returning a value instead of a reference.